server/graphql/asset/src/mutations/insert.rs

```diff
@@ -30,29 +30,20 @@ pub fn insert_asset(
     )?;
 
     // add store_id if not inserting from central server
-    let asset_input;
-    if !CentralServerConfig::is_central_server() {
-        match &input.store_id {
-            Some(input_store_id) => {
-                if input_store_id != store_id {
-                    return Ok(InsertAssetResponse::Error(InsertAssetError {
-                        error: InsertAssetErrorInterface::PermissionError(NoPermissionForThisStore),
-                    }));
-                }
-                asset_input = input;
-            }
-            None => {
-                asset_input = {
-                    InsertAssetInput {
-                        store_id: Some(store_id.to_owned()),
-                        ..input
-                    }
-                }
-            }
+    let asset_input = match &input.store_id {
+        Some(input_store_id)
+            if !CentralServerConfig::is_central_server() && input_store_id != store_id =>
```

Comment:
This is cool I didn't know you could chain logic statements in match. 

But I'm also wondering if this refactor is necessary if this is doing the same thing? Am I understanding correctly that the resulting logic is unchanged?

Reply:
Thanks @fergie-nz! 

It's a slight bit different. 

Before we don't allocate a store_id if inserting from central server. But now we do.

- Inserting an asset from Cold Chain page should use the store_ID regardless if central server or not
   - it should check if they have the permission to update the store to insert an asset 
- Inserting an asset from Manage page should use the selected store_id from the dropdown

The updated logic almost meets the newly desired behaviour, I noticed that the current logic currently doesn't do the permission check on a central server. Going to update the logic